File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon_species/PokemonSpecies.java
package pt.rfsfernandes.model.pokemon_species;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;
import pt.rfsfernandes.data.local.PokemonSpeciesTypeConverters;

/**
 * Class PokemonSpecies created at 1/17/21 20:00 for the project PoketDex
 * By: rodrigofernandes
 */
@Entity
@TypeConverters(PokemonSpeciesTypeConverters.class)
public class PokemonSpecies {
  @PrimaryKey
  private int id;
  @SerializedName("flavor_text_entries")
  private List<FlavourEntries> mFlavourEntriesList;

  public PokemonSpecies(int id, List<FlavourEntries> flavourEntriesList) {
    this.id = id;
    mFlavourEntriesList = flavourEntriesList;
  }

  public int getId() {
    return id;
  }

  public void setId(int id) {
    this.id = id;
  }

  public List<FlavourEntries> getFlavourEntriesList() {
    return mFlavourEntriesList;
  }

  public void setFlavourEntriesList(List<FlavourEntries> flavourEntriesList) {
    mFlavourEntriesList = flavourEntriesList;
  }
}

File: PoketDex/app/src/main/java/pt/rfsfernandes/model/pokemon_species/PokemonSpecies.kt
package pt.rfsfernandes.model.pokemon_species

import androidx.room.Entity
import androidx.room.TypeConverters
import pt.rfsfernandes.data.local.PokemonTypeConverters
import pt.rfsfernandes.model.SimpleModelData
import com.google.gson.annotations.SerializedName
import pt.rfsfernandes.data.local.PokemonSpeciesTypeConverters
import pt.rfsfernandes.model.pokemon_species.FlavourEntries
import androidx.room.PrimaryKey

/**
 * Class PokemonSpecies created at 1/17/21 20:00 for the project PoketDex
 * By: rodrigofernandes
 */
@Entity
@TypeConverters(PokemonSpeciesTypeConverters::class)
class PokemonSpecies(@field:PrimaryKey var id: Int, @field:SerializedName("flavor_text_entries") var flavourEntriesList: List<FlavourEntries>)